tools/testsys/src/aws_resources.rs

```diff
@@ -290,28 +383,31 @@ impl AwsK8s {
                     secrets: Some(self.config.secrets.clone()),
                     capabilities: None,
                 },
-                destruction_policy: DestructionPolicy::OnDeletion,
+                destruction_policy: DestructionPolicy::OnTestSuccess,
             },
             status: None,
         };
         Ok(Crd::Resource(ec2_resource))
     }
 
+    #[allow(clippy::too_many_arguments)]
     fn sonobuoy_crd(
         &self,
         test_name_suffix: &str,
         cluster_name: &str,
+        test_type: &TestType,
+        ec2_resource_name_suffix: &str,
         sonobuoy_mode: SonobuoyMode,
         depends_on: Option<Vec<String>>,
         testsys_images: &TestsysImages,
     ) -> Result<Crd> {
-        let ec2_resource_name = format!("{}-instances", cluster_name);
+        let ec2_resource_name = format!("{}-instances-{}", cluster_name, ec2_resource_name_suffix);
```

Comment:
I feel like `clippy` is on to something here. 😀 

It seems like there's a type that could be factored out, e.g.
```
struct CrdInput {
    cluster_name: ...,
    test_type: ...,
    suffix: ...,
    testsys_images: ...,
}
```

With options for the fields that aren't common, or else a constructor that takes a base `CrdInput` and turns it into a `SonobuoyCrdInput`.

This is not a blocker but the module feels overdue for refactoring, and that might be better done before the next round of features.